entrypoint.sh

```diff
@@ -28,8 +32,43 @@ if [ "$BUILD_LATEX" = true ] ; then
 fi
 apk add $PACKAGES
 
-# run "regular" doxygen
-doxygen $1
+
+# Tests if the $4 is either 'true' or 'false'.
+if [ ! -z $4 ]; then
+    echo "Failing on warnings is enabled."
+    FAIL_ON_WARNINGS=true
+else
+    echo "Failing on warnings disabled."
+    FAIL_ON_WARNINGS=0
+fi
+
+if [ "$FAIL_ON_WARNINGS" = true ]; then
+    doxygen $1 2> warnings.txt
+
+    # if $5 is non-empty, apply filter.
+    if [ ! -z "$5" ]; then
+      grep "$5" warnings.txt > filtered-out-warnings.txt
```

Comment:
Could you look into a way in which we can filter out multiline warnings?

Reply:
If I interpret the doxygen documentation correctly, there shouldn't really be multiline warnings.
And I assume that in multiple stages of the script... (e.g., using `wc -l` for the amount of warnings...).

Possibly, one could filter out that message, or, alternatively, remove the newline before such a message.

Reply:
If I interpret the doxygen documentation correctly, there shouldn't really be multiline warnings.
And I assume that in multiple stages of the script... (e.g., using `wc -l` for the amount of warnings...).

Possibly, one could filter out that message, or, alternatively, remove the newline before such a message.

Reply:
Sth. like this:
https://stackoverflow.com/questions/26188316/if-a-line-starts-with-a-certain-character-remove-the-previous-newline

Reply:
Could we could use WARN_LOGFILE as a [command line option](https://www.doxygen.nl/manual/faq.html#faq_cmdline) and then filter from that file? I've looked at doxygen's source and *all* warnings are prefixed with `warning:` so we can split based on that and then filter the resulting strings.

Reply:
1. Using `WARN_LOGFILE`: Possible, but I assume it's the same result as using `2> warnings.txt`.
2. I also thought about using `warning:`. If someone provides a custom WARN_FORMAT, everything fails, anyways :) But using the default, It should work fine.

Reply:
Could use sed to pull the option out of the doxyfile like we do with the LaTeX option.

Reply:
There can be multi line warnings in doxygen like:
```
cakephp-4.2.8/src/Collection/Iterator/FilterIterator.php:67: warning: no uniquely matching class member found for 
  Cake\Collection\Iterator\FilterIterator::unwrap()
```
or
```
mesa-21.1.5/src/gallium/frontends/clover/core/program.cpp:28: warning: no uniquely matching class member found for 
  program::program(clover::context &ctx, std::string &&source, clover::program::il_type il_type)
Possible candidates:
  'clover::program::program(clover::context &ctx, std::string &&il, enum il_type il_type)' at line 43 of file mesa-21.1.5/src/gallium/frontends/clover/core/program.hpp
  'clover::program::program(clover::context &ctx, const ref_vector< device > &devs={}, const std::vector< module > &binaries={})' at line 46 of file mesa-21.1.5/src/gallium/frontends/clover/core/program.hpp
  'clover::program::program(const program &prog)=delete' at line 50 of file mesa-21.1.5/src/gallium/frontends/clover/core/program.hpp
```
but there are more  (some have even a return in them as in the input part there is a return).

Note that there is also  the setting `WARN_FORMAT` it it is possible to insert. In the past I used:
```
 WARN_FORMAT = "#DOXY1#$file#DOXY2#$line#DOXY3#$text#DOXY4#"
```
so I could filter with awk.

Reply:
Be aware of the fact that in the newer versions of doxygen (as of 1.9.0) there is the possibility `FAIL_ON_WARNINGS` with the setting `WARN_AS_ERROR`.:
> WARN_AS_ERROR
>
> If the WARN_AS_ERROR tag is set to YES then doxygen will immediately stop when a warning
is encountered. If the WARN_AS_ERROR tag is set to FAIL_ON_WARNINGS then doxygen will continue
running as if WARN_AS_ERROR tag is set to NO, but at the end of the doxygen process doxygen will return
with a non-zero status.
> Possible values are: NO, YES and FAIL_ON_WARNINGS.
> The default value is: NO.